src/meta/src/rpc/ddl_controller.rs

```diff
@@ -1798,17 +1809,56 @@ impl DdlController {
         let id = stream_job.id();
         let expr_context = stream_ctx.to_expr_context();
 
+        // check if performing drop table connector
+        let mut drop_table_associated_source_id = None;
+        if let StreamingJob::Table(None, _, _) = &stream_job {
+            let old_table_associate_source_id = self
+                .metadata_manager
+                .get_table_catalog_by_ids(vec![id])
```

Comment:
Nits, can we add a function to get `optional_associated_source_id` directly for better performance?